src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractMethodOrPropertyOrEventSymbolReferenceFinder.cs

```diff
@@ -17,40 +17,43 @@ protected AbstractMethodOrPropertyOrEventSymbolReferenceFinder()
         {
         }
 
-        protected override async Task<IEnumerable<ISymbol>> DetermineCascadedSymbolsAsync(
-            TSymbol symbol,
+        protected override async Task<IEnumerable<SymbolAndProjectId>> DetermineCascadedSymbolsAsync(
+            SymbolAndProjectId<TSymbol> symbolAndProjectId,
             Solution solution,
             IImmutableSet<Project> projects,
             CancellationToken cancellationToken)
         {
             // Static methods can't cascade.
+            var symbol = symbolAndProjectId.Symbol;
             if (!symbol.IsStatic)
             {
                 if (symbol.ContainingType.TypeKind == TypeKind.Interface)
                 {
                     // We have an interface method.  Find all implementations of that method and
                     // cascade to them.
-                    return await SymbolFinder.FindImplementationsAsync(symbol, solution, projects, cancellationToken).ConfigureAwait(false);
+                    return await SymbolFinder.FindImplementationsAsync(symbolAndProjectId, solution, projects, cancellationToken).ConfigureAwait(false);
                 }
                 else
                 {
                     // We have a normal method.  Find any interface methods that it implicitly or
                     // explicitly implements and cascade down to those.
-                    var interfaceMembersImplemented = await SymbolFinder.FindImplementedInterfaceMembersAsync(symbol, solution, projects, cancellationToken).ConfigureAwait(false);
+                    var interfaceMembersImplemented = await SymbolFinder.FindImplementedInterfaceMembersAsync(
+                        symbolAndProjectId, solution, projects, cancellationToken).ConfigureAwait(false);
 
                     // Finally, methods can cascade through virtual/override inheritance.  NOTE(cyrusn):
                     // We only need to go up or down one level.  Then, when we're finding references on
                     // those members, we'll end up traversing the entire hierarchy.
-                    var overrides = await SymbolFinder.FindOverridesAsync(symbol, solution, projects, cancellationToken).ConfigureAwait(false);
+                    var overrides = await SymbolFinder.FindOverridesAsync(
+                        symbolAndProjectId, solution, projects, cancellationToken).ConfigureAwait(false);
 
-                    var overriddenMember = symbol.OverriddenMember();
-                    return overriddenMember == null
+                    var overriddenMember = symbolAndProjectId.WithSymbol(symbol.OverriddenMember());
```

Comment:
Are we guaranteed that the overridden member is defined in the same project as the original symbol? Or is that not what the `ProjectId` represents?